linera-sdk/src/ethereum.rs

```diff
@@ -5,31 +5,46 @@
 
 use std::fmt::Debug;
 
-use ethers::providers::{JsonRpcClient, ProviderError};
-use serde::{de::DeserializeOwned, Serialize};
+use async_graphql::scalar;
+use async_trait::async_trait;
+use linera_ethereum::{client::JsonRpcClient, common::EthereumServiceError};
+use serde::{Deserialize, Serialize};
 
 use crate::contract::wit::contract_system_api;
 
 /// A wrapper for a URL that implements `JsonRpcClient` and uses the JSON oracle to make requests.
-#[derive(Debug)]
+#[derive(Debug, Default, Deserialize, Serialize)]
 pub struct EthereumClient {
     /// The URL of the JSON-RPC server, without the method or parameters.
     pub url: String,
 }
+scalar!(EthereumClient);
 
-#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
-#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl EthereumClient {
+    /// Creates a new `EthereumClient` from an URL.
+    pub fn new(url: String) -> Self {
+        Self { url }
+    }
+}
+
+/// The Ethereum type for a single event
+pub type EthereumDataType = linera_ethereum::common::EthereumDataType;
+
+/// The Ethereum type for an event
+pub type EthereumEvent = linera_ethereum::common::EthereumEvent;
+
+/// The U256 type from Ethereum
+pub type U256 = alloy::primitives::U256;
```

Comment:
These should probably be `pub use`, not `type` aliases?

Reply:
That is related to the above problem. I could not make things run with the `ethereum` feature.

Reply:
Does it work now?
I don't see why `pub use` wouldn't work here.

Reply:
No it does not work. That is the `use linera_sdk::ethereum::U256` does not match if I use the ethereum feature. I will look further into it as it appears important.

Reply:
It's not a blocker from my side, but I don't understand: Replacing these line with `pub use` should work.

```suggestion
pub use linera_ethereum::common::{EthereumDataType, EthereumEvent};
pub use alloy::primitives::U256;
```

Reply:
I found my error and corrected it.